examples/kvstore_38/main.rs

```diff
@@ -0,0 +1,237 @@
+
+    fn poll_ready(&mut self, _cx: &mut Context<'_>) -> Poll<Result<(), Self::Error>> {
+        Poll::Ready(Ok(()))
+    }
+
+    fn call(&mut self, req: Request) -> Self::Future {
+        tracing::info!(?req);
+
+        let rsp = match req {
+            // handled messages
+            Request::Info(_) => Response::Info(self.info()),
+            Request::Query(query) => Response::Query(self.query(query.data)),
+            // Note: https://github.com/tendermint/tendermint/blob/v0.38.x/spec/abci/abci%2B%2B_tmint_expected_behavior.md#adapting-existing-applications-that-use-abci
+            Request::PrepareProposal(prepare_prop) => Response::PrepareProposal(PrepareProposal {
+                txs: prepare_prop.txs,
+            }),
+            Request::ProcessProposal(..) => {
+                Response::ProcessProposal(response::ProcessProposal::Accept)
+            }
+            Request::ExtendVote(vote) => Response::ExtendVote(self.extend_vote(vote)),
+            Request::VerifyVoteExtension(vote) => {
+                Response::VerifyVoteExtension(self.verify_vote(vote))
+            }
+            Request::FinalizeBlock(block) => Response::FinalizeBlock(self.finalize_block(block)),
+            Request::Commit => Response::Commit(self.commit()),
+
+            // unhandled messages
+            Request::Flush => Response::Flush,
+            Request::Echo(_) => Response::Echo(Default::default()),
+            Request::InitChain(_) => Response::InitChain(Default::default()),
+            Request::CheckTx(_) => Response::CheckTx(Default::default()),
+            Request::ListSnapshots => Response::ListSnapshots(Default::default()),
+            Request::OfferSnapshot(_) => Response::OfferSnapshot(Default::default()),
+            Request::LoadSnapshotChunk(_) => Response::LoadSnapshotChunk(Default::default()),
+            Request::ApplySnapshotChunk(_) => Response::ApplySnapshotChunk(Default::default()),
+        };
+        tracing::info!(?rsp);
+        async move { Ok(rsp) }.boxed()
+    }
+}
+
+impl KVStore {
+    fn info(&self) -> response::Info {
+        response::Info {
+            data: "tower-abci-kvstore-example".to_string(),
+            version: "0.1.0".to_string(),
+            app_version: 1,
+            last_block_height: self.height.into(),
+            last_block_app_hash: self.app_hash.to_vec().try_into().unwrap(),
+        }
+    }
+
+    fn query(&self, query: Bytes) -> response::Query {
+        let key = String::from_utf8(query.to_vec()).unwrap();
+        let (value, log) = match self.store.get(&key) {
+            Some(value) => (value.clone(), "exists".to_string()),
+            None => ("".to_string(), "does not exist".to_string()),
+        };
+
+        response::Query {
+            log,
+            key: key.into_bytes().into(),
+            value: value.into_bytes().into(),
+            ..Default::default()
+        }
+    }
+
+    fn execute_tx(&mut self, tx: Bytes) -> ExecTxResult {
```

Comment:
Execute TX feels odd in the examples here since it's a byproduct of `finalize_block` for 0.38?

Reply:
Would it be clearer if it was inline in the `finalize_block` method?

Reply:
Ah I think I understand what you mean, to clarify: this is exactly an example for `0.38`. There are others for `v0.37` and `v0.34` respectively, but not in this PR diff.

Reply:
Gotcha, I think I understand the purpose better now. Think this is perfectly fine. Was just confused by a non abci++  method in the example here